<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>nlSandboxClass</class>
 <widget class="QMainWindow" name="nlSandboxClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>2080</width>
    <height>1663</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Ignored">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>0</width>
    <height>0</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>16777215</width>
    <height>16777215</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>nlSandbox</string>
  </property>
  <property name="dockOptions">
   <set>QMainWindow::AllowTabbedDocks|QMainWindow::AnimatedDocks|QMainWindow::ForceTabbedDocks|QMainWindow::VerticalTabs</set>
  </property>
  <widget class="QWidget" name="dummy">
   <property name="minimumSize">
    <size>
     <width>0</width>
     <height>0</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>0</width>
     <height>0</height>
    </size>
   </property>
   <widget class="widget_previewdock" name="previewDock">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>657</y>
      <width>1280</width>
      <height>720</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <property name="minimumSize">
     <size>
      <width>1280</width>
      <height>720</height>
     </size>
    </property>
    <property name="maximumSize">
     <size>
      <width>1280</width>
      <height>720</height>
     </size>
    </property>
    <property name="floating">
     <bool>true</bool>
    </property>
    <property name="features">
     <set>QDockWidget::DockWidgetFloatable|QDockWidget::DockWidgetMovable</set>
    </property>
    <property name="allowedAreas">
     <set>Qt::NoDockWidgetArea</set>
    </property>
    <property name="windowTitle">
     <string>Preview</string>
    </property>
    <widget class="widget_preview" name="previewWidget">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="minimumSize">
      <size>
       <width>0</width>
       <height>0</height>
      </size>
     </property>
     <property name="maximumSize">
      <size>
       <width>1280</width>
       <height>720</height>
      </size>
     </property>
     <layout class="QGridLayout" name="gridLayout"/>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>2080</width>
     <height>24</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>File</string>
    </property>
    <widget class="QMenu" name="menuFolder">
     <property name="title">
      <string>Folder</string>
     </property>
     <addaction name="actionMedia"/>
     <addaction name="actionMedia_Scene"/>
     <addaction name="actionMedia_Shader"/>
     <addaction name="actionMedia_Model"/>
     <addaction name="actionMedia_Path"/>
     <addaction name="actionExe"/>
    </widget>
    <addaction name="menuFolder"/>
    <addaction name="actionMake"/>
    <addaction name="actionSaveCurrentScene"/>
   </widget>
   <widget class="QMenu" name="menuWindow">
    <property name="title">
     <string>Window</string>
    </property>
    <addaction name="actionClose_Open_preview"/>
    <addaction name="actionClose_Open_path"/>
    <addaction name="actionClose_Open_SceneEdit"/>
    <addaction name="actionClose_Open_Timeline"/>
   </widget>
   <widget class="QMenu" name="menuPreview">
    <property name="title">
     <string>Preview</string>
    </property>
    <addaction name="actionPreview_1_1"/>
    <addaction name="actionPreview_1_2"/>
    <addaction name="actionPreview_1_4"/>
    <addaction name="actionPreview_goto_origin"/>
   </widget>
   <addaction name="menu"/>
   <addaction name="menuWindow"/>
   <addaction name="menuPreview"/>
  </widget>
  <widget class="TimeLineDock" name="dockTimeline">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Ignored" vsizetype="Fixed">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="minimumSize">
    <size>
     <width>1030</width>
     <height>80</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>524287</width>
     <height>80</height>
    </size>
   </property>
   <property name="features">
    <set>QDockWidget::DockWidgetFloatable|QDockWidget::DockWidgetMovable</set>
   </property>
   <property name="windowTitle">
    <string>TimeLine</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_4">
    <layout class="QGridLayout" name="gridLayout_5">
     <item row="1" column="1">
      <widget class="QSlider" name="sliderTimeline">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="styleSheet">
        <string notr="true"/>
       </property>
       <property name="maximum">
        <number>3600</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="tickPosition">
        <enum>QSlider::TicksBothSides</enum>
       </property>
       <property name="tickInterval">
        <number>1000</number>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QSpinBox" name="timeStartTL">
       <property name="maximum">
        <number>1000000000</number>
       </property>
       <property name="singleStep">
        <number>16</number>
       </property>
      </widget>
     </item>
     <item row="1" column="2">
      <widget class="QSpinBox" name="timeEndTL">
       <property name="maximum">
        <number>1000000000</number>
       </property>
       <property name="singleStep">
        <number>16</number>
       </property>
      </widget>
     </item>
     <item row="1" column="3">
      <widget class="QSpinBox" name="timeNow">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="buttonSymbols">
        <enum>QAbstractSpinBox::NoButtons</enum>
       </property>
       <property name="maximum">
        <number>1000000000</number>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="SceneEditDock" name="dockSceneEdit">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="minimumSize">
    <size>
     <width>735</width>
     <height>271</height>
    </size>
   </property>
   <property name="floating">
    <bool>false</bool>
   </property>
   <property name="features">
    <set>QDockWidget::DockWidgetFloatable|QDockWidget::DockWidgetMovable</set>
   </property>
   <property name="windowTitle">
    <string>SceneEdit</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>4</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents">
    <layout class="QGridLayout" name="gridLayout_2">
     <item row="0" column="1" rowspan="2">
      <layout class="QGridLayout" name="gridLayout_8">
       <property name="verticalSpacing">
        <number>7</number>
       </property>
       <item row="1" column="0">
        <widget class="QLabel" name="sceneEditPropLabel0">
         <property name="minimumSize">
          <size>
           <width>50</width>
           <height>0</height>
          </size>
         </property>
         <property name="text">
          <string>Prop0</string>
         </property>
        </widget>
       </item>
       <item row="14" column="0" colspan="4">
        <widget class="QListWidget" name="sceneNodePropList">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Ignored" vsizetype="Ignored">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>50</width>
           <height>0</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>1000</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="layoutDirection">
          <enum>Qt::LeftToRight</enum>
         </property>
         <property name="editTriggers">
          <set>QAbstractItemView::DoubleClicked</set>
         </property>
         <property name="movement">
          <enum>QListView::Static</enum>
         </property>
         <property name="resizeMode">
          <enum>QListView::Fixed</enum>
         </property>
         <property name="selectionRectVisible">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="0" column="3">
        <widget class="QSpinBox" name="sceneEditValveNum">
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>1024</number>
         </property>
        </widget>
       </item>
       <item row="1" column="3">
        <widget class="QDoubleSpinBox" name="sceneEditProp0Float">
         <property name="decimals">
          <number>5</number>
         </property>
         <property name="minimum">
          <double>-4096.000000000000000</double>
         </property>
         <property name="maximum">
          <double>4096.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.010000000000000</double>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="sceneEditPropLabel1">
         <property name="text">
          <string>Prop1</string>
         </property>
        </widget>
       </item>
       <item row="2" column="3">
        <widget class="QDoubleSpinBox" name="sceneEditProp1Float">
         <property name="decimals">
          <number>5</number>
         </property>
         <property name="minimum">
          <double>-4096.000000000000000</double>
         </property>
         <property name="maximum">
          <double>4096.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.010000000000000</double>
         </property>
        </widget>
       </item>
       <item row="3" column="0">
        <widget class="QLabel" name="sceneEditPropLabel2">
         <property name="text">
          <string>Prop2</string>
         </property>
        </widget>
       </item>
       <item row="3" column="3">
        <widget class="QDoubleSpinBox" name="sceneEditProp2Float">
         <property name="decimals">
          <number>5</number>
         </property>
         <property name="minimum">
          <double>-4096.000000000000000</double>
         </property>
         <property name="maximum">
          <double>4096.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.010000000000000</double>
         </property>
        </widget>
       </item>
       <item row="4" column="0">
        <widget class="QLabel" name="sceneEditPropLabel3">
         <property name="text">
          <string>Prop3</string>
         </property>
        </widget>
       </item>
       <item row="4" column="3">
        <widget class="QDoubleSpinBox" name="sceneEditProp3Float">
         <property name="decimals">
          <number>5</number>
         </property>
         <property name="minimum">
          <double>-4096.000000000000000</double>
         </property>
         <property name="maximum">
          <double>4096.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.010000000000000</double>
         </property>
        </widget>
       </item>
       <item row="5" column="0">
        <widget class="QLabel" name="sceneEditPropLabel4">
         <property name="text">
          <string>Prop4</string>
         </property>
        </widget>
       </item>
       <item row="5" column="3">
        <widget class="QDoubleSpinBox" name="sceneEditProp4Float">
         <property name="decimals">
          <number>5</number>
         </property>
         <property name="minimum">
          <double>-4096.000000000000000</double>
         </property>
         <property name="maximum">
          <double>4096.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.010000000000000</double>
         </property>
        </widget>
       </item>
       <item row="6" column="0">
        <widget class="QCheckBox" name="chkStopThisNode">
         <property name="text">
          <string>動作</string>
         </property>
        </widget>
       </item>
       <item row="6" column="2">
        <widget class="QCheckBox" name="chkRenderAxis">
         <property name="text">
          <string>座標</string>
         </property>
        </widget>
       </item>
       <item row="6" column="3">
        <widget class="QLabel" name="labelFps">
         <property name="text">
          <string>FPS:</string>
         </property>
        </widget>
       </item>
       <item row="15" column="0" colspan="4">
        <widget class="QTextEdit" name="textComment">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Ignored" vsizetype="Ignored">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>100</width>
           <height>10</height>
          </size>
         </property>
        </widget>
       </item>
       <item row="0" column="2">
        <widget class="QComboBox" name="comboBox_scene">
         <property name="minimumSize">
          <size>
           <width>80</width>
           <height>0</height>
          </size>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QSpinBox" name="sceneTotalTime">
         <property name="maximum">
          <number>1000000000</number>
         </property>
         <property name="singleStep">
          <number>16</number>
         </property>
        </widget>
       </item>
       <item row="1" column="2">
        <widget class="QComboBox" name="sceneEditProp0Combo"/>
       </item>
       <item row="2" column="2">
        <widget class="QComboBox" name="sceneEditProp1Combo"/>
       </item>
       <item row="3" column="2">
        <widget class="QComboBox" name="sceneEditProp2Combo"/>
       </item>
       <item row="4" column="2">
        <widget class="QComboBox" name="sceneEditProp3Combo"/>
       </item>
       <item row="5" column="2">
        <widget class="QComboBox" name="sceneEditProp4Combo"/>
       </item>
      </layout>
     </item>
     <item row="0" column="0" rowspan="2">
      <widget class="SceneEditor" name="sceneEditor">
       <property name="verticalScrollBarPolicy">
        <enum>Qt::ScrollBarAlwaysOff</enum>
       </property>
       <property name="horizontalScrollBarPolicy">
        <enum>Qt::ScrollBarAlwaysOff</enum>
       </property>
       <property name="backgroundBrush">
        <brush brushstyle="NoBrush">
         <color alpha="0">
          <red>0</red>
          <green>0</green>
          <blue>0</blue>
         </color>
        </brush>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="dockPath">
   <property name="minimumSize">
    <size>
     <width>705</width>
     <height>432</height>
    </size>
   </property>
   <property name="features">
    <set>QDockWidget::DockWidgetFloatable|QDockWidget::DockWidgetMovable</set>
   </property>
   <property name="allowedAreas">
    <set>Qt::AllDockWidgetAreas</set>
   </property>
   <property name="windowTitle">
    <string>CurveEdit</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>4</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_5">
    <layout class="QGridLayout" name="gridLayout_4">
     <item row="0" column="1">
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <item>
        <layout class="QGridLayout" name="gridLayout_9">
         <property name="sizeConstraint">
          <enum>QLayout::SetDefaultConstraint</enum>
         </property>
         <item row="0" column="2">
          <widget class="QListWidget" name="pathList">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Expanding">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>200</width>
             <height>0</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>100</width>
             <height>16777215</height>
            </size>
           </property>
          </widget>
         </item>
         <item row="1" column="2">
          <widget class="QPushButton" name="resetPath">
           <property name="maximumSize">
            <size>
             <width>100</width>
             <height>100</height>
            </size>
           </property>
           <property name="text">
            <string>ResetPath</string>
           </property>
          </widget>
         </item>
         <item row="2" column="2">
          <widget class="QRadioButton" name="radioCurveLiner">
           <property name="maximumSize">
            <size>
             <width>100</width>
             <height>100</height>
            </size>
           </property>
           <property name="text">
            <string>Liner</string>
           </property>
          </widget>
         </item>
         <item row="3" column="2">
          <widget class="QRadioButton" name="radioCurveCatmul">
           <property name="maximumSize">
            <size>
             <width>100</width>
             <height>16777215</height>
            </size>
           </property>
           <property name="text">
            <string>Catmul</string>
           </property>
          </widget>
         </item>
         <item row="4" column="2">
          <widget class="QWidget" name="widget" native="true">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="widget_curve" name="pathWidget" native="true">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>400</width>
             <height>0</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>16777215</width>
             <height>16777215</height>
            </size>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_6"/>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="CurveEditor" name="graphicsView"/>
         </item>
        </layout>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="dockLog">
   <property name="features">
    <set>QDockWidget::DockWidgetFloatable|QDockWidget::DockWidgetMovable</set>
   </property>
   <property name="allowedAreas">
    <set>Qt::NoDockWidgetArea</set>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_2">
    <layout class="QGridLayout" name="gridLayout_3">
     <item row="0" column="0">
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <widget class="QLineEdit" name="logFilter"/>
       </item>
       <item>
        <widget class="QToolButton" name="toolButton">
         <property name="text">
          <string>...</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item row="1" column="0">
      <widget class="QListView" name="logList">
       <property name="selectionMode">
        <enum>QAbstractItemView::NoSelection</enum>
       </property>
       <property name="flow">
        <enum>QListView::TopToBottom</enum>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <action name="actionExe">
   <property name="text">
    <string>Exe</string>
   </property>
  </action>
  <action name="actionMedia_Scene">
   <property name="text">
    <string>Media/Scene</string>
   </property>
  </action>
  <action name="actionMedia_Path">
   <property name="text">
    <string>Media/Path</string>
   </property>
  </action>
  <action name="actionMedia">
   <property name="text">
    <string>Media</string>
   </property>
  </action>
  <action name="actionMedia_Model">
   <property name="text">
    <string>Media/Model</string>
   </property>
  </action>
  <action name="actionClose_Open_preview">
   <property name="text">
    <string>preview</string>
   </property>
  </action>
  <action name="actionMake">
   <property name="text">
    <string>Make!</string>
   </property>
   <property name="shortcut">
    <string comment="F5" extracomment="F5"/>
   </property>
   <property name="shortcutContext">
    <enum>Qt::ApplicationShortcut</enum>
   </property>
   <property name="autoRepeat">
    <bool>false</bool>
   </property>
  </action>
  <action name="actionClose_Open_path">
   <property name="text">
    <string>Path</string>
   </property>
  </action>
  <action name="actionClose_Open_SceneEdit">
   <property name="text">
    <string>SceneEdit</string>
   </property>
  </action>
  <action name="actionClose_Open_Timeline">
   <property name="text">
    <string>Timeline</string>
   </property>
  </action>
  <action name="actionPreview_1_1">
   <property name="text">
    <string>1/1</string>
   </property>
  </action>
  <action name="actionPreview_1_2">
   <property name="text">
    <string>1/2</string>
   </property>
  </action>
  <action name="actionPreview_1_4">
   <property name="text">
    <string>1/4</string>
   </property>
  </action>
  <action name="actionMedia_Shader">
   <property name="text">
    <string>Media/Shader</string>
   </property>
  </action>
  <action name="actionPreview_goto_origin">
   <property name="text">
    <string>原点に移動</string>
   </property>
  </action>
  <action name="actionSaveCurrentScene">
   <property name="text">
    <string>SaveNowScene</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>widget_preview</class>
   <extends>QWidget</extends>
   <header>widget_preview.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>widget_curve</class>
   <extends>QWidget</extends>
   <header>widget_curve.h</header>
   <container>1</container>
   <slots>
    <slot>onChangePathIdx(int)</slot>
    <slot>onChangeCurveMode(int)</slot>
    <slot>onon(QString)</slot>
    <slot>onChangeCurveName(QString)</slot>
   </slots>
  </customwidget>
  <customwidget>
   <class>widget_previewdock</class>
   <extends>QDockWidget</extends>
   <header>widget_previewdock.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>TimeLineDock</class>
   <extends>QDockWidget</extends>
   <header>timelinedock.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>SceneEditDock</class>
   <extends>QDockWidget</extends>
   <header>sceneeditdock.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>SceneEditor</class>
   <extends>QGraphicsView</extends>
   <header>sceneeditor.h</header>
  </customwidget>
  <customwidget>
   <class>CurveEditor</class>
   <extends>QGraphicsView</extends>
   <header>curveeditor.h</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>sceneTotalTime</tabstop>
  <tabstop>sceneEditValveNum</tabstop>
  <tabstop>sceneEditProp0Float</tabstop>
  <tabstop>sceneEditProp1Float</tabstop>
  <tabstop>sceneEditProp2Float</tabstop>
  <tabstop>sceneEditProp3Float</tabstop>
  <tabstop>sceneEditProp4Float</tabstop>
  <tabstop>chkStopThisNode</tabstop>
  <tabstop>chkRenderAxis</tabstop>
  <tabstop>textComment</tabstop>
  <tabstop>sceneEditor</tabstop>
  <tabstop>timeEndTL</tabstop>
  <tabstop>timeNow</tabstop>
  <tabstop>timeStartTL</tabstop>
  <tabstop>sliderTimeline</tabstop>
  <tabstop>pathList</tabstop>
  <tabstop>resetPath</tabstop>
  <tabstop>radioCurveLiner</tabstop>
  <tabstop>radioCurveCatmul</tabstop>
  <tabstop>sceneNodePropList</tabstop>
 </tabstops>
 <resources>
  <include location="nlsandbox.qrc"/>
 </resources>
 <connections/>
</ui>
